mvcc/backend/backend.go

```diff
@@ -163,6 +165,20 @@ func (b *backend) ForceCommit() {
 }
 
 func (b *backend) Snapshot() Snapshot {
+	stopc, donec := make(chan struct{}), make(chan struct{})
+	go func() {
+		defer close(donec)
+		start := time.Now()
+		for {
+			select {
+			case <-time.After(snapshotWarningTimeout):
+				plog.Warningf("snapshotting is taking more than %v seconds to finish [started at %v]", time.Since(start).Seconds(), start)
```

Comment:
I would really recommend printing out the total time taken in addition to this warning, otherwise one can't tell the difference between 31 and 301 seconds...